src/System.Runtime.Extensions/src/System/IO/StreamWriter.cs

```diff
@@ -246,10 +246,10 @@ private void Flush(bool flushStream, bool flushEncoder)
             if (!_haveWrittenPreamble)
             {
                 _haveWrittenPreamble = true;
-                byte[] preamble = _encoding.GetPreamble();
+                ReadOnlySpan<byte> preamble = _encoding.Preamble;
                 if (preamble.Length > 0)
                 {
-                    _stream.Write(preamble, 0, preamble.Length);
+                    _stream.Write(preamble);
```

Comment:
FYI, this is just a temporary issue, but I've/we've not yet overridden the span-based methods on FileStream, so temporarily this won't be as efficient as it could be.  Not something to block on, though.